server.go

```diff
@@ -296,12 +332,13 @@ func Serve(opts *ServeConfig) {
 	logger.Debug("plugin address", "network", listener.Addr().Network(), "address", listener.Addr().String())
 
 	// Output the address and service name to stdout so that the client can bring it up.
-	fmt.Printf("%d|%d|%s|%s|%s%s\n",
+	fmt.Printf("%d|%d|%s|%s|%s|%s%s\n",
```

Comment:
I _think_ this is OK, since it looks like the client never parses that last `extra` field at the end (`extra` includes the `|` if it exists). If it does turn out to be an issue, maybe we could bump the core protocol number.

The issue I had with passing the server's cert via the environment, is that it would contain the private key, and the process environment is generally world readable on unix systems.

Reply:
When grpc support was first added to the protocol we had an inadvertent break here with a similar change. I'm not sure if the parsing of the handshake was made more robust later, but I'd suggest testing that carefully since I distinctly remember us thinking the _other_ change (adding the wire protocol specifier to the end) would be fine for a similar reason, and yet it was not. :confounded:

Reply:
The discussion about what I was alluding to my last comment is in hashicorp/terraform#15756.

Reply:
I think this was made more robust after we introduced gRPC, but that's just something in my mind that I remember. I agree that we should test carefully. :)

Reply:
In my latest changes I removed the `extra` field as it was never used by the client, which in essences leaves the protocol the same, just making use of the last field for the cert.